test/jdk/java/awt/print/PrinterJob/PrintTextTest.java

```diff
@@ -240,319 +259,204 @@ private static Font getPhysicalFont() {
         return physicalFont;
     }
 
-    public PrintTextTest(String page, Font font, AffineTransform gxTx,
-                         boolean fm) {
-        this.page = page;
-        textFont = font;
-        this.gxTx = gxTx;
-        this.useFM = fm;
-        setBackground(Color.white);
-    }
-
-    public static AttributedCharacterIterator getIterator(String s) {
-        return new AttributedString(s).getIterator();
-    }
-
-    static String orient(PageFormat pf) {
-        if (pf.getOrientation() == PageFormat.PORTRAIT) {
-            return "Portrait";
-        } else {
-            return "Landscape";
+    private static class PrintText extends Component implements Printable {
+
+        protected Font textFont;
+        protected AffineTransform gxTx;
+        protected String page;
+        protected boolean useFM;
+        protected int preferredSize;
+
+        public PrintText(String page, Font font, AffineTransform gxTx, boolean fm, int size) {
+            this.page = page;
+            this.textFont = font;
+            this.gxTx = gxTx;
+            this.useFM = fm;
+            this.preferredSize = size;
+            setBackground(Color.WHITE);
         }
-    }
-
-    public int print(Graphics g, PageFormat pf, int pageIndex) {
-
-        Graphics2D g2d = (Graphics2D)g;
-        g2d.translate(pf.getImageableX(),  pf.getImageableY());
-        g.drawString(page+" "+orient(pf),50,20);
-        g.translate(0, 25);
-        paint(g);
-        return PAGE_EXISTS;
-    }
-
-    public Dimension getMinimumSize() {
-        return getPreferredSize();
-    }
 
-    public Dimension getPreferredSize() {
-        return new Dimension(preferredSize, preferredSize);
-    }
-
-    public void paint(Graphics g) {
-
-        /* fill with white before any transformation is applied */
-        g.setColor(Color.white);
-        g.fillRect(0, 0, getSize().width, getSize().height);
-
-
-        Graphics2D g2d = (Graphics2D) g;
-        if (gxTx != null) {
-            g2d.transform(gxTx);
-        }
-        if (useFM) {
-            g2d.setRenderingHint(RenderingHints.KEY_FRACTIONALMETRICS,
-                                 RenderingHints.VALUE_FRACTIONALMETRICS_ON);
+        public static AttributedCharacterIterator getIterator(String s) {
+            return new AttributedString(s).getIterator();
         }
 
-        g.setFont(textFont);
-        FontMetrics fm = g.getFontMetrics();
-
-        String s;
-        int LS = 30;
-        int ix=10, iy=LS+10;
-        g.setColor(Color.black);
-
-        s = "drawString(String str, int x, int y)";
-        g.drawString(s, ix, iy);
-        if (!textFont.isTransformed()) {
-            g.drawLine(ix, iy+1, ix+fm.stringWidth(s), iy+1);
+        static String orient(PageFormat pf) {
```

Comment:
```suggestion
        private static String orient(PageFormat pf) {
```
I'd make it `private` too.